### PR TITLE
fix: use signed stacks account in transaction

### LINKED ISSUE #4599: Fix recognition of active account during Stacks contract execution
The first account in the wallet is shown as the active one for signing, both in the modal's header and in the "switch account" drawer when opened, even when a different account has been connected to the site e.g. https://stacks.gamma.io/

https://github.com/leather-wallet/extension/assets/28991/3000e1cf-3383-410c-9498-503836d30fd1

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
> Try out this version of Leather — [Extension build](https://github.com/leather-wallet/extension/actions/runs/7813295142), [Test report](https://leather-wallet.github.io/playwright-reports/fix/use-correct-signed-account)

Closes #4599

Fix account name and avatar for transactions:

https://github.com/leather-wallet/extension/assets/22010816/1371fca1-8aac-4ff0-b3a1-e8dcf092faa6

Note: noticed a weird bg color when picking an account for sign in

### CODE REVIEW

**Comment by markmhendrickson** [comment]
Curious why the popup window appears so wide for you? What browser is this?

**Comment by edgarkhanzadian** [comment]
@markmhendrickson it's Arc related. It remembers dimensions of the previous popup window

Chrome:

https://github.com/leather-wallet/extension/assets/22010816/0b0e790c-cbf2-451a-8815-84c55257d100


Arc:


https://github.com/leather-wallet/extension/assets/22010816/73472a3a-4a99-4e59-9cac-d17af6dfacdc

File: src/app/pages/home/components/home.layout.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Stack } from 'leather-styles/jsx';
 
 import { AccountInfoCard } from './account-info-card';
 
-type HomeLayoutProps = Record<'currentAccount' | 'children', React.ReactNode>;
+type HomeLayoutProps = Record<'children', React.ReactNode>;
```
Comment (kyranjamie) [inline src/app/pages/home/components/home.layout.tsx:8]: We have a `HasChildren` helper type we could replace this with

**Review by kyranjamie — APPROVED** [review]
LGTM.

**Comment by markmhendrickson** [comment]
Let's get this merged

**Comment by pete-watters** [comment]
I'm going to close this PR as I made the same changes in https://github.com/leather-wallet/extension/pull/5152 to avoid rebasing issues

**Pull request closed without merging**